integration-cli/docker_cli_build_test.go

```diff
@@ -5565,6 +5565,22 @@ func (s *DockerSuite) TestBuildNetContainer(c *check.C) {
 	c.Assert(strings.TrimSpace(host), check.Equals, "foobar")
 }
 
+func (s *DockerSuite) TestBuildWithExtraHost(c *check.C) {
+	testRequires(c, DaemonIsLinux)
+
+	name := "testbuildwithextrahost"
+	buildImageSuccessfully(c, name,
+		withBuildFlags(
+			"--add-host", "foo:127.0.0.1",
+			"--add-host", "bar:127.0.0.1",
```

Comment:
I know we're in the design phase and this is premature, but if I don't ask now I'll probably forget....
can you add some negative tests? For example `--add-host myhost:` or `--add-host myhost` and ones with bad IPs. And ones with ipv6